pkg/engine/task/task_apply.go

```diff
@@ -64,6 +65,11 @@ func apply(ro []runtime.Object, c client.Client) ([]runtime.Object, error) {
 		key, _ := client.ObjectKeyFromObject(r)
 		existing := r.DeepCopyObject()
 
+		// if CRD we need to clear then namespace from the copy
```

Comment:
Do you mind wrapping this and `client.ObjectKeyFromObject(r)` call into a common method `GetObjectKey` or smth.?

Reply:
yeah... conversations with Tom last night indicated the need to make this a general utility... although this is the only place in code this is used currently